gcloud/storage/blob.py

```diff
@@ -510,6 +511,13 @@ def upload_from_file(self, file_obj, rewind=False, size=None,
         if self.chunk_size is not None:
             upload.chunksize = self.chunk_size
 
+            if total_bytes is None:
+                upload.strategy = RESUMABLE_UPLOAD
+        elif self.chunk_size is None and total_bytes is None:
```

Comment:
We know unconditionally that that `self.chunk_size` is None here (or we would have taken the prior branch).